### ISSUE TITLE
first point getting cropped

### ISSUE THREAD

**Issue by rupeshjainearnin** [patrykandpatrick/vico#248]
first point getting cropped

**Module**

- [x] `compose`
- [ ] `views`
- [ ] `compose-m2`
- [ ] `compose-m3`
- [x] `core` (select if none of the above apply)

**To reproduce**
Create a chart with 4 data point and a round point shape

    val pointShape = shapeComponent(
                shape = Shapes.pillShape,
                color = blue,
                strokeColor = blue
                strokeWidth = 1.dp,
            )

**Expected behavior**
The first point should not cropped

**Screenshots**

<img width="400" alt="Screenshot 2023-04-25 at 4 44 47 PM" src="https://user-images.githubusercontent.com/78571048/234665161-a9a3a681-2940-4b2c-bc26-e6db4a601f9b.png">

**Android versions**
All versions

**Additional context**
NA

**Comment by Gowsky** [comment]
Hello @rupeshjainearnin, regarding your expected behavior, how would you expect the library to handle horizontal scrolling without the points getting clipped? And how should it deal with points that could cover axis labels?

**Comment by rupeshjainearnin** [comment]
Any of the following fixes should work

1. (Preferred)is there a way to add padding so that the starting point doesn't get cropped, we start drawing the labels after the padding
2. draw the point over the Y axis- I think this is better than cropping

**Comment by Gowsky** [comment]
I've discussed possible solutions with @patrickmichalik, and we've decided to add a possibility to add padding to ticks on x-axes.

We can't remove the clip completely, as it could lead to a `point` covering content on y-axes. The clip would have to be moved somewhere further to work with scrolling, and make the behaviour consistent with horizontal scrolling enabled or disabled.

If you're short on time, I'd recommend creating a subclass of `LineChart` and overriding `drawChartInternal()` function to modify, or remove the clip. Most of the API is open for subclasses and modifications.
